server/middleware/barcode/barcodeAuthorisationMiddleware.ts

```diff
@@ -8,6 +13,24 @@ export default function barcodeAuthorisationMiddleware(): RequestHandler {
       return res.redirect('/link/request-link')
     }
 
-    return next()
+    return verify(
+      req.session.slmToken,
+      config.barcodeTokenPublicKey,
+      { algorithms: ['RS256'] },
+      (err: VerifyErrors) => {
+        if (err) {
+          const errorMessage = `Found an invalid JWT: ${req.session.slmToken} which caused error: ${err}`
+          logger.warn(errorMessage)
+          req.session.validSlmToken = false
+          req.session.slmToken = undefined
+          return next(
+            createError(401, 'invalid JWT', {
+              code: 'EBADJWT',
+            })
```

Comment:
Normally the user is signed out due to a CSRF error, so I pinched the same method of forcing a sign out as the csurf lib to keep it consistent.